packages/components/src/DateTimePickers/LegacyDateTimePickers/DateTime/date-extraction.js

```diff
@@ -256,8 +279,12 @@ function dateTimeToStr(date, time, options) {
 
 	const { hours, minutes, seconds } = time;
 	try {
-		const timeInSeconds = timeToSeconds(hours, minutes, seconds);
+		const timeInSeconds = timeToSeconds(hours, minutes, seconds, options.hybridMode);
 		const fullDate = setSeconds(date, timeInSeconds);
+		if (!hours && !minutes && !seconds && options.hybridMode) {
+			// if in hybrid mode and time are empty we don't want time in text input
```

Comment:
This if is hard to understand, if we need a comment here it's maybe it's not as simple as it's will do.

Maybe create a named const to help the understanding :

`const isTimeNeeded = !hours && !minutes && !seconds && options.hybridMode;`